src/main/java/com/cms/example/cms/feature/geo/UpazilaRepository.java

```diff
@@ -1,17 +1,31 @@
 package com.cms.example.cms.feature.geo;
 
+import com.cms.example.cms.entities.District;
 import com.cms.example.cms.entities.Upazila;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
 import java.util.Optional;
 
 @Repository
 public interface UpazilaRepository extends JpaRepository<Upazila, Long> {
 
     @Query("SELECT u FROM Upazila u LEFT JOIN FETCH u.district WHERE u.upazilaId = :upazilaId")
     Optional<Upazila> findByIdWithDetails(@Param("upazilaId") Long upazilaId);
+
+    @Query("SELECT u FROM Upazila u WHERE " +
+            "(:upazilaId IS NULL OR u.upazilaId = :upazilaId) AND " +
+            "(:name IS NULL OR u.name ILIKE %:name%) AND " +
+            "(:nameLocal IS NULL OR u.nameLocal ILIKE %:nameLocal%) AND " +
+            "(:active IS NULL OR u.active = :active) AND " +
+            "(:districtId IS NULL OR u.district.districtId = :districtId)")
```

Comment:
aei vabe nested na kore, first join kore rekhe dibi, then oi join use korbi

```suggestion
    @Query("SELECT u FROM Upazila u LEFT JOIN District d WHERE " +
            "(:upazilaId IS NULL OR u.upazilaId = :upazilaId) AND " +
            "(:name IS NULL OR u.name ILIKE %:name%) AND " +
            "(:nameLocal IS NULL OR u.nameLocal ILIKE %:nameLocal%) AND " +
            "(:active IS NULL OR u.active = :active) AND " +
            "(:districtId IS NULL OR d.districtId = :districtId)")
```

Reply:
changed according to the comment.